pkg/kubectl/apply/merge/strategic_visitor.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package merge
+
+import "k8s.io/kubernetes/pkg/kubectl/apply"
+
+// strategicVisitor delegates merging fields to other visitor implementations
+// based on the merge strategy preferred by the field.
+type strategicVisitor struct {
+	options Options
+	merge   mergeVisitor
+	replace replaceVisitor
+}
+
+// createStrategicVisitor returns a new strategicVisitor
+func createStrategicVisitor(options Options) *strategicVisitor {
+	v := &strategicVisitor{
```

Comment:
nit: why don't you just initialize all `strategicVisitor` fields at once?

Reply:
Because of a circular dependency on `v`

Reply:
OK, I see it now :-)